doc/src/advanced.rst

```diff
@@ -30,9 +30,23 @@ connections when they are not needed anymore::
       c.login("bar@foo.org", "passwd")
       c.select_folder("INBOX")
 
-Watching a mailbox asynchronously using idle
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-TODO
+Watching a mailbox using idle
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+The IDLE extension allows an IMAP server to notify a client when something
+changes in a mailbox. It can be used as an alternative to polling to receive
```

Comment:
Maybe emphasise the "selected" mailbox notion (e.g. a sent message appended in INBOX/Sent won't end the IDLE if you're watching the INBOX)

Reply:
Done